python/dgl/utils/internal.py

```diff
@@ -912,6 +912,35 @@ def _fn(*args, **kwargs):
 
 def recursive_apply(data, fn, *args, **kwargs):
```

Comment:
I will make this function internally used by `DataLoader`. For user code, have a separate `hetero_apply` which only accepts dictionary and apply function on each value.